text_handler.py

```diff
@@ -24,16 +22,20 @@ def __init__(self, bot: Bot):
         self.bot = bot
 
     async def handle(self, message: types.Message):
-        if not await self.__isProvideService(message):
+        if not await self.__is_provide_service(message):
             return
         text = message.text
-        if text and (' ' not in text and '\n' not in text) \
+        if text and (text.startswith("http")) \
```

Comment:
looks like a pretty complicated **if** chain, I suggest to move the contents of it into a function. for example function called `__is_video_for_download` (maybe you can think of a better name)
then it will look like this
```
        if self.__is_video_for_download(text):
            await types.ChatActions.typing()
            await self.__handle_music_grabber(message)
```

Reply:
yes, pylint also indicate that. another solution could be using regular expression, but needs to import its library. I will try to move it into a function.

Reply:
if you have a problem that you want to solve with regular expressions then you have two problems :)
regular expressions are good, but they are usually pretty complicated and hard to read for regular developers. So in my opinion they should be used only to solve very specific problems that couldn't be solved any other way. If you used some complicated regular expression here it would take me like 30 minutes to figure out what each key symbol in that expression means